### ISSUE TITLE
Question By A ML Beginner

### ISSUE THREAD

**Issue by Low-www-wig** [neeharperi/DeepKNNDefense#3]
Sorry for bothering. I am a beginner in ML. I have some problems running the project directly. In "extractFeatures.py", I found that the project didn't have a directory named "modelCheckPoints". I want to create the directory manually and use the DenseNet121 model downloaded from Internet. But I received error reporting telling me that the key doesn't match. Could you please help you solve the problem?

**Comment by neeharperi** [comment]
I've updated the README to answer this question. Please try again and let me know if you encounter any further issues.

**Comment by Low-www-wig** [comment]
Sorry for bothering again. I tried it out with the new readme file. I successfully connected and unzipped the compression package and got the model file. But when I put the file into use, there were new error files that I couldn't understand. It seems like there were still problems with the dicts of models. They might still not match. Could you please help me again?
![image](https://user-images.githubusercontent.com/63444656/134533500-0935211b-a1c4-4e4d-bfb4-d4830c13efc4.png)
![image](https://user-images.githubusercontent.com/63444656/134533570-c80bd358-3b76-4a23-89d9-ef73be5e4d51.png)

**Comment by neeharperi** [comment]
I wrote this code using a multi-GPU setup so the models are saved with DataParallel. If you load the model, you'll see that all the keys are there, they just have a "module" label prepended. If you are not running this code on a multi-GPU setup. I'd recommend modifying the state dictionaries. This [link ](https://discuss.pytorch.org/t/solved-keyerror-unexpected-key-module-encoder-embedding-weight-in-state-dict/1686/4) might be useful.

**Comment by Low-www-wig** [comment]
Sorry for bothering you with the problem that RoyMikeJiang has met before. I've browsed the forum and use OrderedDict to remove the prefix of "module.". Ant then I've also encoutered the key error like this:
![image](https://user-images.githubusercontent.com/63444656/134691197-89ee1b97-3457-4343-b661-ee636a288ec7.png)
I'm not familiar with the details of the code and I don't have multi-GPUs. Could u please help me?

**Comment by neeharperi** [comment]
After you modify the state dictionary to remove the module keyword, make sure you preserve the structure of the dictionary in the original pth file. Try setting CheckPoint["net"] = new_state_dict

**Comment by MasterVChicken** [comment]
I'm really sorry for bothering again. I hated myself for being stupid. But after spending so many time on solving the problem, I still couldn't solve it. I found the question lies on that I should move away the string "module." in the value rather than in the key. Could you please help me with some code?
